phoenix-core/src/main/java/org/apache/phoenix/log/QueryLogger.java

```diff
@@ -92,9 +92,17 @@ public boolean isSynced(){
         }
     };
 
-    public static QueryLogger getInstance(PhoenixConnection connection, boolean isSystemTable) {
-        if (connection.getLogLevel() == LogLevel.OFF || isSystemTable || ThreadLocalRandom.current()
-                .nextDouble() > connection.getLogSamplingRate()) { return NO_OP_INSTANCE; }
+    public static QueryLogger getInstance(PhoenixConnection connection, boolean isSystemTable,
```

Comment:
Should we rename this to `createInstance` so that it doesn't feel like we are returning a singleton?

Reply:
Good idea, changed.